src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -255,14 +256,34 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         }
         else if (WI_IsFlagSet(buttonState, MouseButtonState::IsRightButtonDown))
         {
-            // Try to copy the text and clear the selection
-            const auto successfulCopy = CopySelectionToClipboard(shiftEnabled, nullptr);
-            _core->ClearSelection();
-            if (_core->CopyOnSelect() || !successfulCopy)
+            if (_core->Settings().RightClickContextMenu())
             {
-                // CopyOnSelect: right click always pastes!
-                // Otherwise: no selection --> paste
-                RequestPasteTextFromClipboard();
+                auto bufferText = _core->SelectedText(true);
+                std::wstring singleString = L"";
+                for (const auto& line : bufferText)
+                {
+                    if (!singleString.empty())
+                    {
+                        singleString += L"\r\n";
+                    }
+                    singleString += line;
+                };

```

Comment:
yknow, I might back this out. I think in my original prototype I bubbled the text up to the hosting app (the page), but I don't think that's really needed anymore...


Oh wait I used that for the "Search the web..." menu entry, so that it could display "Search the web for 'the thing I selected'" in the menu. 

Yea I'll back that out for now.